src/web_buffer.h

```diff
@@ -1,77 +1,266 @@
+/**
+ * Set buffer not cacheable
+ *
+ * @param wb The Buffer.
+ */
 #define buffer_no_cacheable(wb) do { (wb)->options |= WB_CONTENT_NO_CACHEABLE; if((wb)->options & WB_CONTENT_CACHEABLE)    (wb)->options &= ~WB_CONTENT_CACHEABLE;  (wb)->expires = 0; } while(0)
 
+/**
+ * Get the length in bytes of the content of a web buffer.
+ *
+ * @param wb The Buffer.
+ * @return length of `wb`
+ */
 #define buffer_strlen(wb) ((wb)->len)
+/**
+ * Get the content of web buffer `wb`
+ *
+ * The returned string is always null terminated.
+ *
+ * @param wb The buffer.
+ * @return content of `wb`
+ */
 extern const char *buffer_tostring(BUFFER *wb);
 
+/**
+ * Ensure `buffer` has `needed_free_size` free bytes.
+ *
+ * Check if buffer has `needed_free_size`. If not increase the buffer.
+ *
+ * @param buffer to check
+ * @param needed_free_size of `buffer`
+ * 
+ */
 #define buffer_need_bytes(buffer, needed_free_size) do { if(unlikely((buffer)->size - (buffer)->len < (size_t)(needed_free_size))) buffer_increase((buffer), (size_t)(needed_free_size)); } while(0)
 
+/**
+ * Flush web_buffer `wb`
+ *
+ * Mark buffer as empty.
+ *
+ * @param wb Buffer to flush.
+ */
 #define buffer_flush(wb) wb->buffer[(wb)->len = 0] = '\0'
+/**
+ * Reset web_buffer `wb`
+ *
+ * Mark buffer as empty and reset metadata.
+ *
+ * @param wb Buffer to reset. 
+ */
 extern void buffer_reset(BUFFER *wb);
 
+/**
+ * Add `txt` to the end of web buffer `wb`.
+ *
+ * @param wb The buffer.
+ * @param txt String to add.
+ */
 extern void buffer_strcat(BUFFER *wb, const char *txt);
+/**
+ * Add `value` to the end of web buffer `wb`.
+ *
+ * @param wb The buffer.
+ * @param value to add
+ */
 extern void buffer_rrd_value(BUFFER *wb, calculated_number value);
 
+/**
+ * Add a date to web buffer `wb`.
+ *
+ * Format: `YYYY-MM-DD hh:mm:ss` (i.e. 2014-04-01 03:28:20)
+ *
+ * @param wb The buffer.
+ * @param year of date
+ * @param month of date
+ * @param day of date
+ * @param hours of date
+ * @param minutes of date
+ * @param seconds of date
+ */
 extern void buffer_date(BUFFER *wb, int year, int month, int day, int hours, int minutes, int seconds);
+/**
+ * Add a date to web buffer `wb`.
+ *
+ * Format: `Date(YYYY,MM,DD,hh,mm,ss)` (i.e. Date(2014,04,01,03,28,20))
+ *
+ * @param wb The buffer.
+ * @param year of date
+ * @param month of date
+ * @param day of date
+ * @param hours of date
+ * @param minutes of date
+ * @param seconds of date
+ */
 extern void buffer_jsdate(BUFFER *wb, int year, int month, int day, int hours, int minutes, int seconds);
 
+/**
+ * Initialize a empty web buffer.
+ *
+ * @param size Initial size of the buffer.
+ * @return the initialized buffer.
+ */
 extern BUFFER *buffer_create(size_t size);
+/**
+ * Free a web buffer.
+ *
+ * @param b The buffer.
+ */
 extern void buffer_free(BUFFER *b);
+/**
+ * Ensure `buffer` has `needed_free_size` free bytes.
+ *
+ * Use buffer_need_bytes() instead. It is quicker.
+ *
+ * \todo Do not extern this.
```

Comment:
> I don't follow. Do not extern buffer_free()? Why?

Because buffer_need_bytes() provides the same functionality but is quicker. Tell me if I am wrong.